tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -2342,6 +2451,38 @@ def _validate_fit_extra_kwargs(self, kwargs, extra_valid_keys=None):
 
         return kwargs_sanitized
 
+    def _prune_data_features(self, train_features: pd.DataFrame, other_features: pd.DataFrame, is_labeled: bool):
+        if self.sample_weight is not None:
+            if self.sample_weight in train_features:
+                train_features.remove(self.sample_weight)
+            if self.sample_weight in other_features:
+                other_features.remove(self.sample_weight)
+        if self._learner.groups is not None and is_labeled:
+            train_features.remove(self._learner.groups)
+
+        return train_features, other_features
+
+    def _validate_pseudo_data(self, train_data: pd.DataFrame, pseudo_data: pd.DataFrame):
```

Comment:
I find "pseudolabeled" to be much clearer than the shorter prefix "pseudo". 
"pseudo_data" seems quite vague to me as a variable name compared to "pseudolabeled_data"

Reply:
Why can't this method simply call `self._validate_fit_data()` somewhere inside of it, instead of duplicating the code from that other method?

Reply:
so the issue is that we don't know if incoming data is labeled or not. That determines where it would go in `self._validate_fit_data()` if it is labeled, it should go into the `tuning_data` parameter, if not it should go into the `unlabled_data` parameter in the `_validate_fit_data()` signature. If the incoming data isn't in pandas format we can't check if the data is labeled or not, `_validate_pseudo_data` does this and handles both cases. There's a little redundancy, but the main thing is checking if data is labeled or not and checking it accordingly if data is not labeled and you pass it into `_validate_fit_data()` `tuning_data` param it throws and error and same with labeled data into the `unlabeled_data` parameter